### PR TITLE
Implemented Browsing in BrowseStoragesController 

### PR DESCRIPTION
**Pull request by sanket407**
Implemented Browsing under /storages for GET requests with Accept : text/html header 
1. If requested path is directory : Correctly displays directory listing inside . Works for all 4 repository types.
2. If requested path is file : 
For Maven, Nuget, Raw : Works correctly i.e. Downloads the requested file.

For NPM: Current implementation of npm doesn't seem to support downloading package.json file . So it doesn't work here too.

Due to path conflict with logic of NugetArtifactController, while making Get request for foll. methods please use their resp. produces Content type in your accept header if you directly wish to request to those methods.
1.downloadPackage - MediaType.APPLICATION_OCTET_STREAM
2. getPackage - MediaType.APPLICATION_OCTET_STREAM
3. greet - MediaType.TEXT_PLAIN

For other GETs , they should be working without Accept header

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseStoragesController.java [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
 
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
 import io.swagger.annotations.ApiOperation;
 import io.swagger.annotations.ApiParam;
 import io.swagger.annotations.ApiResponse;
 import io.swagger.annotations.ApiResponses;
 import java.io.IOException;
+import java.util.Arrays;
+import javax.servlet.RequestDispatcher;
```
Comment (fuss86) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseStoragesController.java]: unused import

all javax.servlet.* neither

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseStoragesController.java [hunk 2]
```diff
@@ -69,4 +87,158 @@ public void browseRepositories(@ApiParam(value = "The storageId", required = tru
...
+    @ApiOperation(value = "Used to browse inside repositories", position = 1)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = ""),
+                            @ApiResponse(code = 404, message = "Requested path not found."),
+                            @ApiResponse(code = 500, message = "Server Error."),
+                            @ApiResponse(code = 503, message = "Repository not in service currently.")})
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(value = { "{storageId}/{repositoryId}/{path:.+}" },
+                    method = {RequestMethod.GET},
+                    headers = {"user-agent=Maven/*"})
+    public void browseMavenRepositoriesInternal(@ApiParam(value = "The storageId", required = true)
```
Comment (fuss86) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseStoragesController.java]: > browseMavenRepositoriesInternal
> browseNugetRepositoriesInternal
> browseNpmRepositoriesInternal
> browseRawRepositoriesInternal

are we going to add similar method for each new repository layout ? Can't we just introduce one method with `headers = {"user-agent"}`  (not sure if `headers = {"user-agent"}`   is necessary ?)

**Comment by carlspring** [comment]
@sanket407 : 

Have you tried executing the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests) against this (with both the Maven and NuGet tests enabled)?

**Comment by sbespalov** [comment]
@sanket407 @carlspring I understand correctly that we must close this request because there is [Implemented browsing in BrowseController under /browse #575](https://github.com/strongbox/strongbox/pull/575)

**Comment by carlspring** [comment]
@sanket407 ,

Thanks for your work. This has been merged as the other pull.

**Pull request closed without merging**